src/assets/highlight-links.js

```diff
@@ -0,0 +1,34 @@
+// eslint-disable-next-line spaced-comment
+/// <reference types="chrome"/>
+
+// Listens for a request from LinkService and respondes with all of the links
+// from the contents of the active page.
+chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
+  try {
+    if (request.message === 'highlight link') {
+      const color = highlightLink(request.linkData, request.newColor);
+      sendResponse({
+        success: true,
+        prevColor: color
```

Comment:
nit: Consider "originalColor" instead of "prevColor". We only handle two colors, the original and the highlight color. PrevColor might suggest that we can switch between more than two colors.

Reply:
Yeah this is a great point, I spent Friday trying to experiment with a better way of communicating this information in general, so I think I'm going to rework a lot of this anyway. The main issue I've been dealing with is how to make sure that someone doesn't have two instances of the extension running. The main benefit here was that if someone already highlighted the link in another instance, then it simply wouldn't highlight in this one. I was trying to add some protection to only allow one instance of the extension to communicate with the content script, but it looks like the issue is that messages from two different instances aren't actually distinguishable to the content script. So maybe we should just go forward assuming people won't open two instances on the same page for now?

Reply:
I see your point. I can see it might be useful for user to open two extension instances to achieve different view. For example, one to group by host and the other view is to group by status. Maybe instead of changing the background color manually, we need to create a custom css class (e.g. link-diver-on-highlight). So we can simply add/remove it when toggling highlight.